website/src/content/blog/2025-03-31-typespec-1-0-release/type_spec_1_0_release.md

```diff
@@ -0,0 +1,202 @@
+    @body todoitems: TodoItem[];
+  }
+
+  @post
+  op createTodoItem(@body body: CreateTodoItemRequest): CreateTodoItemResponse;
+
+  model CreateTodoItemResponse {
+    @statusCode statusCode: 201;
+    @body todoitem: TodoItem;
+  }
+}
+
+model TodoItem {
+  id: string;
+  content: string;
+  isCompleted: boolean;
+  labels: string[];
+}
+
+model CreateTodoItemRequest {
+  content: string;
+  labels?: string[];
+}
+```
+
+From this definition, TypeSpec generates foundational server-side code, such as controllers and models. For example, a generated C# controller method might look like this:
+
+```csharp
+[HttpGet]
+[Route("/todoitems")]
+[ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(TodoItem[]))]
+public virtual async Task<IActionResult> GetTodoItems()
+{
+    var result = await TodoItemsOperationsImpl.GetTodoItemsAsync();
+    return Ok(result);
+}
+```
+
+And a corresponding model:
+
+```csharp
+public partial class TodoItem
+{
+    public string Id { get; set; }
+    public string Content { get; set; }
+    public bool IsCompleted { get; set; }
+    public string[] Labels { get; set; }
+}
+```
+
+With the generated code in place, you can quickly integrate it into your project, build the service, and run it. TypeSpec ensures that your API implementation stays perfectly aligned with your specification, saving you time and effort.
+
+- **Note:** The C# service codegen is currently in preview, so there may be gaps in the implementation. We want service codegen to be rock solid when we move from RC to full release, so we encourage you to try it out and tell us if/when it breaks to help us improve this feature.
+
+### Client-side code generation
+
+In addition to server-side code, TypeSpec can generate client-side libraries for multiple languages, including JavaScript, Python, C#, and Java, with support for additional languages like Rust in development.
+
+For example, a generated JavaScript library will allow you to interact with your `todo` API like this:
+
+```javascript
+const client = new TodoServiceClient();
+const todo = await client.createTodoItem({
+  content: "Buy groceries",
+  dueDate: "2025-03-21",
+  isCompleted: false,
+});
+console.log(todo);
+```
+
+TypeSpec simplifies maintenance by allowing seamless updates to server and client code as APIs evolve, enabling you to focus on business logic rather than repetitive tasks.
+
+## Emitters: From TypeSpec to the API format you need
+
+TypeSpec's core value proposition is the ability to define your API once and generate multiple output formats. The 1.0-RC release ships with a stable OpenAPI 3.0 emitter (`@typespec/openapi3`), enabling you to seamlessly integrate TypeSpec into existing OpenAPI workflows and toolchains.
+
+But our emitter story doesn't stop there. The TypeSpec ecosystem includes several additional emitters:
+
+- **OpenAPI 2.0 Emitter**: For teams that still require Swagger/OpenAPI 2.0 compatibility
+- **REST Emitter**: Produces JSON descriptions of REST APIs
```

Comment:
What is this emitter? These should probably include package names and links to NPM.